shm/map.cpp

```diff
@@ -0,0 +1,50 @@
+#include "map.hpp"
+#include <algorithm>
+#include <functional>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <random>
+
+map::map() {
+    auto returnPaired = generateIslandPosition();
+    std::vector<Island> vecofIslands{};
+    map_.reserve(MAX_ISLANDS);
+    for (int i = 0; i < MAX_ISLANDS; ++i) {
+        map_.push_back(std::make_shared<Island>(Coordinates(returnPaired[i].first, returnPaired[i].second)));
+    }
+}
+
+void map::printIslands() const {
+    for (auto& a : map_) {
+        std::cout << a.get()->getPosition().getPositionX() << ' ' << a.get()->getPosition().getPositionY() << '\n';
+    }
+}
+
+std::vector<std::pair<int, int>> map::generateIslandPosition() {
+    std::map<int, int> vecOfPairs{};
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> distrib(1, 16);
```

Comment:
I think this should be a member of the class. Creating a random device based on `mt19937` take time, and also can make some problems, because we have a limited resource of pseudo numbers.